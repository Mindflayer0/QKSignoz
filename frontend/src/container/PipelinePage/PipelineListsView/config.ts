import { ColumnsType } from 'antd/es/table';

import { PipelineColumn, ProcessorColumn } from '.';

export const addPipelinefieldLists = [
	{
		id: 1,
		fieldName: 'Name',
		placeholder: 'Name',
		name: 'name',
	},
	{
		id: 2,
		fieldName: 'Tags',
		placeholder: 'Tags',
		name: 'tags',
	},
	{
		id: 3,
		fieldName: 'Description',
		placeholder: 'Enter description for your pipeline',
		name: 'operators',
	},
];

export const tagInputStyle: React.CSSProperties = {
	width: 78,
	verticalAlign: 'top',
	flex: 1,
};

export const pipelineColumns: ColumnsType<PipelineColumn> = [
	{
		key: 'orderid',
		title: '',
	},
	{
		key: 'name',
		title: 'Pipeline Name',
	},
	{
		key: 'filter',
		title: 'Filters',
	},
	{
		key: 'tags',
		title: 'Tags',
	},
	{
		key: 'updatedAt',
		title: 'Last Edited',
	},
	{
		key: 'updatedBy',
		title: 'Edited By',
	},
];

export const processorColumns: ColumnsType<ProcessorColumn> = [
	{
		key: 'id',
		title: '',
	},
	{
		key: 'text',
		title: '',
	},
];
