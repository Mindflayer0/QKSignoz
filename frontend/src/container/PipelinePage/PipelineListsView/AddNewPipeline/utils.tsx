import React from 'react';

import { pipelineFields } from '../config';
import DescriptionTextArea from './FormFields/DescriptionTextArea';
import FilterSearch from './FormFields/FilterSearch';
import NameInput from './FormFields/NameInput';
// import ProcessorTags from './FormFields/ProcessorTags';

export const renderPipelineForm = (): // setTagsListData: (tags: Array<string>) => void,
// tagsListData?: Array<string>,
Array<JSX.Element> =>
	pipelineFields.map((fieldName) => {
		if (fieldName.id === '1') {
			return <FilterSearch key={fieldName.id} fieldData={fieldName} />;
		}
		if (fieldName.id === '2') {
			return <NameInput key={fieldName.id} fieldData={fieldName} />;
		}
		/* if (fieldName.id === '3') {
			return (
				<ProcessorTags
					key={fieldName.id}
					fieldData={fieldName}
					setTagsListData={setTagsListData}
					tagsListData={tagsListData || []}
				/>
			);
		} */
		return <DescriptionTextArea key={fieldName.id} fieldData={fieldName} />;
	});
