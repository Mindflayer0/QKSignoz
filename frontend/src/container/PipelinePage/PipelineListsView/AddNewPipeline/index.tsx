import { Button, Divider, Form, Modal } from 'antd';
import React, { useCallback, useEffect, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { AppState } from 'store/reducers';
import { PipelineData } from 'types/api/pipeline/def';
import AppReducer from 'types/reducer/app';

import { ActionMode, ActionType } from '../../Layouts/Pipeline';
import { ModalButtonWrapper, ModalTitle } from '../styles';
import { getEditedDataSource, getRecordIndex } from '../utils';
import { renderPipelineForm } from './utils';

function AddNewPipeline({
	isActionType,
	setActionType,
	selectedPipelineData,
	setShowSaveButton,
	setCurrPipelineData,
	currPipelineData,
}: AddNewPipelineProps): JSX.Element {
	const [form] = Form.useForm();
	const { t } = useTranslation('pipeline');
	const { user } = useSelector<AppState, AppReducer>((state) => state.app);
	// const [tagsListData, setTagsListData] = useState<Array<string>>();

	const isEdit = isActionType === 'edit-pipeline';
	const isAdd = isActionType === 'add-pipeline';

	useEffect(() => {
		if (isEdit) {
			// setTagsListData(selectedPipelineData?.tags);
			form.setFieldsValue(selectedPipelineData);
		}
		if (isAdd) {
			// setTagsListData([]);
			form.resetFields();
		}
	}, [form, isEdit, isAdd, selectedPipelineData]);

	const onFinish = (values: PipelineData): void => {
		const newPipeLineData: PipelineData = {
			orderId: currPipelineData.length + 1,
			createdAt: new Date().toISOString(),
			createdBy: user?.name || '',
			name: values.name,
			alias: values.name.replace(/\s/g, ''),
			description: values.description,
			filter: values.filter,
			// tags: tagsListData || [],
			config: [],
			enabled: true,
		};

		if (isEdit && selectedPipelineData) {
			const findRecordIndex = getRecordIndex(
				currPipelineData,
				selectedPipelineData,
				'name',
			);
			const updatedPipelineData: PipelineData = {
				...currPipelineData[findRecordIndex],
				name: values.name,
				description: values.description,
				filter: values.filter,
				// tags: tagsListData || [],
			};

			const editedPipelineData = getEditedDataSource(
				currPipelineData,
				selectedPipelineData,
				'name',
				updatedPipelineData,
			);

			setCurrPipelineData(editedPipelineData);
		}
		if (isAdd) {
			// setTagsListData([]);
			setCurrPipelineData((prevState) => [...prevState, newPipeLineData]);
		}
		setActionType(undefined);
	};

	const onCancelModalHandler = (): void => {
		setActionType(undefined);
	};

	const modalTitle = useMemo(
		(): string =>
			isEdit
				? `${t('edit_pipeline')} : ${selectedPipelineData?.name}`
				: t('create_pipeline'),
		[isEdit, selectedPipelineData?.name, t],
	);

	const onOkModalHandler = useCallback(
		() => setShowSaveButton(ActionMode.Editing),
		[setShowSaveButton],
	);

	const isOpen = useMemo(() => isEdit || isAdd, [isAdd, isEdit]);

	return (
		<Modal
			title={<ModalTitle level={4}>{modalTitle}</ModalTitle>}
			centered
			open={isOpen}
			width={800}
			footer={null}
			onCancel={onCancelModalHandler}
		>
			<Divider plain />
			<Form
				name="add-new-pipeline"
				layout="vertical"
				onFinish={onFinish}
				autoComplete="off"
				form={form}
			>
				{/* {renderPipelineForm(setTagsListData, tagsListData)} */}
				{renderPipelineForm()}
				<Divider plain />
				<Form.Item>
					<ModalButtonWrapper>
						<Button
							key="submit"
							type="primary"
							htmlType="submit"
							onClick={onOkModalHandler}
						>
							{isEdit ? t('update') : t('create')}
						</Button>
						<Button key="cancel" onClick={onCancelModalHandler}>
							{t('cancel')}
						</Button>
					</ModalButtonWrapper>
				</Form.Item>
			</Form>
		</Modal>
	);
}

interface AddNewPipelineProps {
	isActionType: string;
	setActionType: (actionType?: ActionType) => void;
	selectedPipelineData: PipelineData | undefined;
	setShowSaveButton: (actionMode: ActionMode) => void;
	setCurrPipelineData: (
		value: React.SetStateAction<Array<PipelineData>>,
	) => void;
	currPipelineData: Array<PipelineData>;
}

export default AddNewPipeline;
