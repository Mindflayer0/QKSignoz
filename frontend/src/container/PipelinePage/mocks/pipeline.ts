import { PipelineColumn } from '../PipelineListsView/types';

export const configurationVerison = '1.0';

export const pipelineMockData: Array<PipelineColumn> = [
	{
		orderid: 1,
		uuid: '22a588b8-cccc-4a49-94f1-2caa28271315',
		createdAt: '1674205513092137500',
		createdBy: {
			username: 'nitya',
			email: 'nityananda@signoz.io',
		},
		updatedAt: 'Mon Feb 06 2023',
		updatedBy: {
			username: 'nitya',
			email: '',
		},
		version: 'myversion1',
		name: 'Apache common parser',
		alias: 'apachecommonparser',
		enabled: true,
		filter: 'attributes.source == nginx',
		tags: ['server', 'app'],
		operators: [
			{
				id: '1',
				type: 'grok',
				name: 'grok use common',
				output: 'grokcommon',
				pattern: '%{COMMONAPACHELOG}',
				parse_from: 'body',
				parse_to: 'attributes',
			},
			{
				id: '2',
				type: 'move',
				name: 'rename auth',
				output: 'renameauth',
				parse_from: 'attributes.auth',
				parse_to: 'attributes.username',
			},
		],
	},
	{
		orderid: 2,
		uuid: 'f58ce70c-5f41-48bc-949d-a822e2409257',
		version: 'myversion1',
		createdAt: '1674205513092137500',
		createdBy: {
			username: 'nitya',
			email: 'nityananda@signoz.io',
		},
		updatedAt: 'Sun Feb 05 2023',
		updatedBy: {
			username: 'nityananda',
			email: '',
		},
		name: 'myservice trace parser',
		alias: 'myservicetraceparser',
		enabled: true,
		filter: 'attributes.source == myservice',
		tags: ['server', 'app', 'host', 'realse'],
		operators: [
			{
				id: '1',
				type: 'trace_parser',
				name: 'Parse trace details',
				output: 'removeoldxtrace_id',
				trace_id: {
					parse_from: 'attributes.xtrace_id',
				},
				span_id: {
					parse_from: 'attributes.xspan_id',
				},
				trace_flags: {
					parse_from: 'attributes.xtrace_flag',
				},
			},
			{
				id: '2',
				type: 'remove',
				name: 'remove old xtrace_id',
				output: 'removeoldxspan_id',
				field: 'attributes.xtrace_id',
			},
			{
				id: '3',
				type: 'remove',
				name: 'remove old xspan_id',
				output: 'removeoldxtrace_flag',
				field: 'attributes.xspan_id',
			},
			{
				id: '4',
				type: 'remove',
				name: 'remove old xtrace_flag',
				output: 'removeoldxtrace_flag',
				field: 'attributes.xtrace_flag',
			},
		],
	},
];
