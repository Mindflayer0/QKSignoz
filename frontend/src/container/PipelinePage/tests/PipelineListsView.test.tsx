import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { pipelineApiResponseMockData } from '../mocks/pipeline';
import PipelineListsView from '../PipelineListsView';
import { matchMedia } from './AddNewPipeline.test';

beforeAll(() => {
	matchMedia();
});

describe('PipelinePage container test', () => {
	it('should render PipelineListsView section', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<PipelineListsView
							isActionType="add-pipeline"
							setActionType={jest.fn()}
							isActionMode="viewing-mode"
							setActionMode={jest.fn()}
							piplineData={pipelineApiResponseMockData}
							refetchPipelineLists={jest.fn()}
						/>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
