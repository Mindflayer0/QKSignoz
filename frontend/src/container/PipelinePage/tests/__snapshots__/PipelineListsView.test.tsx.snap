// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage test should render PipelinePage 1`] = `
<DocumentFragment>
  .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  margin-left: 5.875rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.c1 {
  color: #DBDBDB;
  font-size: 1.5rem;
}

.c2 {
  color: #DBDBDB;
  font-size: 1.5rem;
}

.c3 {
  color: #DBDBDB;
  font-size: 1.5rem;
}

.c5 {
  color: #8DCFF8;
  font-size: 1.12rem;
}

.c0 {
  margin-top: 3rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
    <div
      class="c0"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-1i536d8"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        class="ant-table-expand-icon-col"
                      />
                      <col
                        style="width: 10px;"
                      />
                      <col
                        style="width: 80px;"
                      />
                      <col
                        style="width: 50px;"
                      />
                      <col
                        style="width: 20px;"
                      />
                      <col
                        style="width: 50px;"
                      />
                      <col
                        style="width: 50px;"
                      />
                      <col
                        style="width: 100px;"
                      />
                      <col
                        style="width: 80px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        />
                        <th
                          class="ant-table-cell"
                        />
                        <th
                          class="ant-table-cell"
                        >
                          Pipeline Name
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Filters
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Tags
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Last Edited
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Edited By
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Actions
                        </th>
                        <th
                          class="ant-table-cell"
                        />
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                        draggable="true"
                      >
                        <td
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-1i536d8"
                            style="background: rgb(22, 104, 220);"
                          >
                            <span
                              class="ant-avatar-string"
                              style="transform: scale(1) translateX(-50%);"
                            >
                              1
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Apache common parser
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          attributes.source == nginx
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                          >
                            server
                          </span>
                          <span
                            class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                          >
                            app
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Mon Feb 06 2023
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            nitya
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space css-dev-only-do-not-override-1i536d8 ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit c1"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="eye"
                                  class="anticon anticon-eye c2"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <span>
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete c3"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="c4"
                          >
                            <span>
                              <button
                                aria-checked="false"
                                class="ant-switch css-dev-only-do-not-override-1i536d8"
                                role="switch"
                                type="button"
                              >
                                <div
                                  class="ant-switch-handle"
                                />
                                <span
                                  class="ant-switch-inner"
                                >
                                  <span
                                    class="ant-switch-inner-checked"
                                  />
                                  <span
                                    class="ant-switch-inner-unchecked"
                                  />
                                </span>
                              </button>
                            </span>
                            <span
                              style="cursor: move;"
                            >
                              <span
                                aria-label="holder"
                                class="anticon anticon-holder c5"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="holder"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M300 276.5a56 56 0 1056-97 56 56 0 00-56 97zm0 284a56 56 0 1056-97 56 56 0 00-56 97zM640 228a56 56 0 10112 0 56 56 0 00-112 0zm0 284a56 56 0 10112 0 56 56 0 00-112 0zM300 844.5a56 56 0 1056-97 56 56 0 00-56 97zM640 796a56 56 0 10112 0 56 56 0 00-112 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                        draggable="true"
                      >
                        <td
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-1i536d8"
                            style="background: rgb(22, 104, 220);"
                          >
                            <span
                              class="ant-avatar-string"
                              style="transform: scale(1) translateX(-50%);"
                            >
                              2
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          myservice trace parser
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          attributes.source == myservice
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                          >
                            server
                          </span>
                          <span
                            class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                          >
                            app
                          </span>
                          <span
                            class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                          >
                            host
                          </span>
                          <span
                            class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                          >
                            realse
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Sun Feb 05 2023
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            nityananda
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space css-dev-only-do-not-override-1i536d8 ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit c1"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="eye"
                                  class="anticon anticon-eye c2"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <span>
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete c3"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="c4"
                          >
                            <span>
                              <button
                                aria-checked="false"
                                class="ant-switch css-dev-only-do-not-override-1i536d8"
                                role="switch"
                                type="button"
                              >
                                <div
                                  class="ant-switch-handle"
                                />
                                <span
                                  class="ant-switch-inner"
                                >
                                  <span
                                    class="ant-switch-inner-checked"
                                  />
                                  <span
                                    class="ant-switch-inner-unchecked"
                                  />
                                </span>
                              </button>
                            </span>
                            <span
                              style="cursor: move;"
                            >
                              <span
                                aria-label="holder"
                                class="anticon anticon-holder c5"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="holder"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M300 276.5a56 56 0 1056-97 56 56 0 00-56 97zm0 284a56 56 0 1056-97 56 56 0 00-56 97zM640 228a56 56 0 10112 0 56 56 0 00-112 0zm0 284a56 56 0 10112 0 56 56 0 00-112 0zM300 844.5a56 56 0 1056-97 56 56 0 00-56 97zM640 796a56 56 0 10112 0 56 56 0 00-112 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
              <div
                class="ant-table-footer"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-link c6"
                  type="button"
                >
                  <span
                    aria-label="plus"
                    class="anticon anticon-plus"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="plus"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <defs>
                        <style />
                      </defs>
                      <path
                        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                      />
                      <path
                        d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                      />
                    </svg>
                  </span>
                  <span>
                    add_new_pipeline
                  </span>
                </button>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1i536d8"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
